### ISSUE TITLE
[SSL ERROR]: shutdown while in init (continually disconnected)

### ISSUE THREAD

**Issue by Coldblackice** [kvirc/KVIrc#2435]
(Build info at bottom)

I've been suddenly and continually been getting disconnected from all IRC servers, with the only cryptic message in the console being "SSL ERROR: shutdown while in init" before a reconnection attempt, successful login, and then shortly after, another SSL ERROR + disconnect. 

This is happening over and over, even after SASL has successfully authenticated. I thought maybe it had to do with the setting to drop connection if SASL couldn't authenticate, thinking maybe the detection was bugging. I disabled that setting and the issue went away for a couple days, but now it's suddenly back again -- nothing at all changed on my parts -- despite the setting still being disabled. What might be going on? Why might it suddenly and spontaneously occur (and then reoccur)?

Here's what the console shows:

```
[6.25.19|17:33:10] Shutting down Ident service (requested action)
[6.25.19|17:33:20] Shutting down Ident service (requested action)
[6.25.19|17:33:21] [SSL ERROR]: shutdown while in init
[6.25.19|17:33:21] [SOCKET ERROR]: Secure Socket Layer error
[6.25.19|17:33:21] Connection terminated [irc.freenode.org (38.229.70.22:6697)]
[6.25.19|17:33:21] The connection terminated unexpectedly. Trying to reconnect...
[6.25.19|17:33:31] Shutting down Ident service (requested action)
```

I've logged it to socket spy, but there isn't much info there, either. I can't even stay connected to servers long enough to discuss this in the channel. Using latest nightly version.

Here's what socket spy logged shows:

```
[17:24:14] :livingstone.freenode.net 367 *** #freenode *!*@216.224.186.70 livingstone.freenode.net 1535075481
[17:24:14] :livingstone.freenode.net 367 *** #freenode $a:alec livingstone.freenode.net 1535075481
[17:24:14] Socket closed
[17:24:26] Socket open
[17:24:26] CAP LS
PING :irc.freenode.org
[17:24:26] :leguin.freenode.net NOTICE * :*** Looking up your hostname...
[17:24:26] :leguin.freenode.net NOTICE * :*** Checking Ident
[17:24:26] :leguin.freenode.net NOTICE * :*** No Ident response
[17:24:27] :leguin.freenode.net NOTICE * :*** Couldn't look up your hostname
[17:24:27] :leguin.freenode.net CAP * LS :account-notify extended-join identify-msg multi-prefix sasl
[17:24:27] CAP REQ :sasl multi-prefix account-notify extended-join
[17:24:27] :leguin.freenode.net 451 * :You have not registered
[17:24:27] :leguin.freenode.net CAP * ACK :sasl multi-prefix account-notify extended-join 
[17:24:27] AUTHENTICATE EXTERNAL
[17:24:27] AUTHENTICATE +
[17:24:27] AUTHENTICATE +
[17:24:28] :leguin.freenode.net 900 * *!unknown@unaffiliated/*** *** :You are now logged in as ***.
[17:24:28] CAP END
[17:24:28] NICK ***
```

Build info:
```
KVIrc 5.0.0 'Aria'

Runtime Info:
System name: Windows Release : 10.0.17763
System version: Windows 10 2019 Pro (x64) (Build 17763)
Architecture: Intel64 Family 6 Model 44 Stepping 2
Qt version: 5.9.7
Qt theme: windowsvista

`Build Info:
Build date: 2019-05-02 00:19:28 UTC
Sources date: 20190502
Revision number: 5.0.0-51-gd8a3ee002
CPU name: AMD64
Build command: C:/Program Files (x86)/CMake/bin/cmake.exe
Build flags: 
   MANDIR=share/man
   CMAKE_INSTALL_PREFIX=C:/projects/kvirc/build/release
   Threads=Win32
Compiler name: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe
Compiler flags: /DWIN32 /D_WINDOWS /W3 /GR /EHsc
Qt version: 5.9.7
Features: IRC, IPv6, Crypt, SSL, IPC, Transparency, Perl, Python, Enchant, Qt5, KVS
OpenSSL version: OpenSSL 1.0.2q 20 Nov 2018
OpenSSL compiler flags: "compiler: cl -utf-8 -FS /MD /Ox -DOPENSSL_THREADS -DDSO_WIN32 -utf-8 -FS -W3 -Gs0 -Gy -nologo -DOPENSSL_SYSNAME_WIN32 -DWIN32_LEAN_AND_MEAN -DL_ENDIAN -DUNICODE -D_UNICODE -D_CRT_SECURE_NO_DEPRECATE -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DRC4_ASM -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DOPENSSL_USE_APPLINK -I. -DOPENSSL_NO_RC5 -DOPENSSL_NO_MD2 -DOPENSSL_NO_SSL2 -DOPENSSL_NO_KRB5 -DOPENSSL_NO_JPAKE -DOPENSSL_NO_WEAK_SSL_CIPHERS -DOPENSSL_NO_DYNAMIC_ENGINE "
OpenSSL built on: reproducible build, date unspecified
```

**Comment by ctrlaltca** [comment]
Crosslinking https://github.com/openssl/openssl/issues/710

**Comment by IceN9ne** [comment]
Try 3f01725 and let me know if you still have the issue. Your logs weren't conclusive as there was a 9 minute discrepancy between console and socket spy output, so I took my best guess (and this probably needed adjusting anyways).

**Comment by Coldblackice** [comment]
@ctrlaltca Thanks, looking into that now. Out of curiosity, how were you able to know about that issue? Were you already aware of it? Or did you search to find it?

@IceN9ne I'll try that, thanks. Nice catch on the logs, I didn't realize. The issue was happening the same, over and over, across multiple servers/contexts. So I must have inadvertently copied from different contexts while bouncing between to see if error messages were different (they were all the same).

**Comment by ctrlaltca** [comment]
Fixed in 3f0172591f580b7209f8528bdb202dde0acfd51d
